File: Abstraction/Pattern Matching/area.ml
let print_area = function
    | [side] -> Printf.printf "Area of square is: %d\n" (side * side)
    | [height; width] -> Printf.printf "Area of rectangle is: %d\n" (height * width)
    | _ -> Printf.printf "Invalid arguments\n"

let _ =
    print_area [5];
    print_area [4; 6]

File: Abstraction/Pattern Matching/area.rs
fn print_area(args: &[i32]) {
    match args {
        [side] => println!("Area of square is: {}", side * side),
        [height, width] => println!("Area of rectangle is: {}", height * width),
        _ => panic!("Invalid number of arguments"),
    }
}

pub fn main() {
    print_area(&[5]);
    print_area(&[4, 6]);
}
